modules/thermalsensor_generic.c

```diff
@@ -0,0 +1,1451 @@
+thermal_sensor_generic_set_temp_func(thermal_sensor_generic_t *self,
+                                     sg_temp_fn cb)
+{
+    self->sg_is_meta = false;
+    self->sg_temp_cb = cb;
+}
+
+/** Set sensor object sensor enable/disable controls
+ *
+ * @param self     sensor object
+ * @param path     file that can be used to enable/disable sensor
+ * @param enable   string to write when enabling sensor
+ * @param disable  string to write when disabling sensor
+ */
+static void
+thermal_sensor_generic_set_mode_control(thermal_sensor_generic_t *self,
+                                        const char *path,
+                                        const char *enable,
+                                        const char *disable)
+{
+    tsg_util_set_string(&self->sg_mode_path,    path);
+    tsg_util_set_string(&self->sg_mode_enable,  enable);
+    tsg_util_set_string(&self->sg_mode_disable, disable);
+}
+
+/** Set sensor object thermal limits
+ *
+ * @param self     sensor object
+ * @param status   thermal status to configure
+ * @param mintem   minimum temperature needed for this status
+ * @param minwait  minimum polling delay while in this status
+ * @param maxwait  maxnimum polling delay while in this status
+ */
+static void
+thermal_sensor_generic_set_limit(thermal_sensor_generic_t *self,
+                                 THERMAL_STATUS status,
+                                 int mintemp, int minwait, int maxwait)
+{
+    if( status >= 0 && status < THERMAL_STATUS_COUNT ) {
+        self->sg_level[status].sl_mintemp = mintemp;
+        self->sg_level[status].sl_minwait = minwait;
+        self->sg_level[status].sl_maxwait = maxwait;
+    }
+}
+
+/** Set sensor object as meta sensor depending on another one
+ *
+ * @param self     sensor object
+ */
+static void
+thermal_sensor_generic_set_depends_on(thermal_sensor_generic_t *self,
+                                      const char *sensor_name)
+{
+    tsg_util_set_string(&self->sg_temp_path, sensor_name);
+    self->sg_is_meta = true;
+    self->sg_temp_cb = tsg_util_read_other;
+}
+
+/** Set sensor object temperature correction offset
+ *
+ * Needed for configuring meta sensors like surface
+ * temperature that is estimated from battery temperature
+ *
+ * @param self     sensor object
+ * @param offs     correction offset [C]
+ */
+static void
+thermal_sensor_generic_set_temp_offs(thermal_sensor_generic_t *self, int offs)
+{
+    self->sg_temp_offs = offs;
+}
+
+/* ========================================================================= *
+ * HOOKS_FOR_THERMAL_OBJECT
+ * ========================================================================= */
+
+/** Hook functions for interfacing via thermal object API */
+static const thermal_sensor_vtab_t thermal_sensor_generic_vtab =
+{
+    .tsv_delete_cb         = thermal_sensor_generic_delete_cb,
+    .tsv_get_name_cb       = thermal_sensor_generic_get_name_cb,
+    .tsv_get_depends_on_cb    = thermal_sensor_generic_get_depends_on_cb,
```

Comment:
Indentation, might want to align `=` with the `=` in the lines above and below (there's space).